server/task/task-memory/src/main/scala/org/apache/james/task/eventsourcing/TaskAggregate.scala

```diff
@@ -64,11 +72,13 @@ class TaskAggregate private(val aggregateId: TaskAggregateId, private val histor
 
   private[eventsourcing] def update(additionalInformation: AdditionalInformation): util.List[Event] =
     (currentStatus match {
-      case Status.IN_PROGRESS => createEvent(AdditionalInformationUpdated(aggregateId, _, additionalInformation))
-      case Status.CANCEL_REQUESTED => createEvent(AdditionalInformationUpdated(aggregateId, _, additionalInformation))
+      case Status.IN_PROGRESS if isAfterLatestAdditionalInformation(additionalInformation.timestamp) => createEvent(AdditionalInformationUpdated(aggregateId, _, additionalInformation))
+      case Status.CANCEL_REQUESTED if isAfterLatestAdditionalInformation(additionalInformation.timestamp) => createEvent(AdditionalInformationUpdated(aggregateId, _, additionalInformation))
       case _ => None
     }).toList.asJava
 
+  private[this] def isAfterLatestAdditionalInformation(timestamp: Instant) : Boolean = latestAdditionalInformationTimestamp.forall(timestamp.isAfter)
```

Comment:
you can put the method into DecisionProjection instead, that would be much more readable